core/src/main/scala/org/apache/spark/scheduler/local/LocalSchedulerBackend.scala

```diff
@@ -59,6 +62,12 @@ private[spark] class LocalEndpoint(
   private val executor = new Executor(
     localExecutorId, localExecutorHostname, SparkEnv.get, userClassPath, isLocal = true)
 
+  private val serializer = new ThreadLocal[SerializerInstance] {
```

Comment:
LocalEndpoint does not support multiple threads,the ThreadLocal is not necessary.